lib/models/package-info-cache/package-info.js

```diff
@@ -220,10 +224,49 @@ class PackageInfo {
     return packages;
   }
 
-  isAddon() {
+  /**
+   * Indicate if this packageInfo is for a project. Should be called only after the project
+   * has been loaded.
+   *
+   * @method isForProject
+   * @return {Boolean} true if this packageInfo is for a Project, false otherwise.
+   */
+  isForProject() {
+    return isProject(this.project);
+  }
+
+  /**
+   * Indicate if this packageInfo is for an Addon.
+   *
+   * @method isForAddon
+   * @return {Boolean} true if this packageInfo is for an Addon, false otherwise.
+   */
+  isForAddon() {
     return isAddon(this.pkg.keywords);
   }
 
+  /**
+   * Indicate if this packageInfo represents a lazy engine.
+   *
+   * @method isForLazyEngine
+   * @return {Boolean} true if this pkgInfo is configured as an engine and the
+   * module this represents has lazyLoading enabled, false otherwise.
+   */
+  isForLazyEngine() {
+    return isEngine(this.pkg.keywords) && isLazyEngine(require(this.realPath));
```

Comment:
we want to use `require(this.addonMainPath)` here; it's not guaranteed that addon entry points are located at `{realPath}/index.js`

Reply:
Good point - thanks!